data/state/accountsDB.go

```diff
@@ -63,10 +67,25 @@ func NewAccountsDB(
 		mutOp:                  sync.RWMutex{},
 		dataTries:              NewDataTriesHolder(),
 		obsoleteDataTrieHashes: make(map[string][][]byte),
-		numCheckpoints:         0,
+		numCheckpoints:         numCheckpoints,
 	}, nil
 }
 
+func getNumCheckpoints(trie data.Trie) uint32 {
+	val, err := trie.Database().Get(numCheckpointsKey)
+	if err != nil {
+		return 0
+	}
+
+	numCheckpoints, err := strconv.Atoi(string(val))
```

Comment:
why not use binary.BigEndian.Uint32() and PutUint32 ? When decoding to uint32, should add a len check as to not have less than 4 bytes as it will panic.

Reply:
Done.